### DISCUSSION TITLE
Angular/Vue style scoped CSS

### DISCUSSION THREAD

**Discussion by msteen** [solidjs/solid#1125] [Q&A]
I am moving from Angular to Solid and would really love to keep Angular's strong CSS isolation. I am aware of CSS modules, but AFAIK this is nothing more than having unique CSS classes, meaning that nested selectors like `.something div` would become `.something-afe8fa div` and still have `div` cascade like normal. I prefer the stronger isolation of Angular that would generate something like `.something[rnd-afe8fa] div[rnd-afe8fa]`, making sure that only the `div` in that particular component is styled as such.

Is it possible to achieve something similar for Solid? Is there already a library for this out there that I am not aware of? And if I would have to make something myself, I imagine I would have to write a Babel plugin or some such so I can rewrite the JSX to include the necessary attributes? If so, are there any resources on this? I did find [a plugin](https://github.com/LXSMNSYC/solid-sfc) building adding features to Solid's JSX that might give me ideas, but I am not sure if is the right approach (or like I said, something might already exist).

**Comment by msteen** [comment]
In my search I somehow missed [Styled JSX](https://github.com/vercel/styled-jsx), for which Solid even has [a wrapper](https://github.com/solidjs/solid-styled-jsx) available. It seems to cover most of what I am looking for.

**Comment by lxsmnsyc** [comment]
If you want scoped styling, maybe https://github.com/LXSMNSYC/solid-styled

**Reply by msteen** [reply]
Thanks for the link! I will give it a try and see how it compares to the other I found.
